main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -116,9 +118,18 @@ object Weeder extends Phase[ParsedAst.Program, WeededAst.Program] {
       for {
         mods <- visitModifiers(mods0, legalModifiers = Set(Ast.Modifier.Inline, Ast.Modifier.Public))
         sigs <- traverse(sigs0)(visitSig)
+        _ <- checkSingleTypeParam(tparams, loc)
       } yield List(WeededAst.Declaration.Class(doc, mods, ident, tparams, sigs.flatten, loc))
   }
 
+  /**
+    * Verifies the size of `tparams0` is exactly 1.
+    */
+  private def checkSingleTypeParam(tparams0: WeededAst.TypeParams, loc: SourceLocation): Validation[Unit, WeederError] = tparams0 match {
```

Comment:
*If* we decide to go with single param type classes then I think this can just be enforced in the syntax.

Reply:
Yes, that makes sense.